### PR TITLE
Add newestOnTop prop on ToastProvider

### LINKED ISSUE #172: Add prop to allow reversal of the order of toast items. For instance NEWEST first e.t.c
https://github.com/jossmac/react-toast-notifications/blob/469bd6ac4c29bed458191b193f465179139547ac/src/ToastProvider.js#L116

### PR DESCRIPTION
**Pull request by ochomoswill**
Add functionality to allow having the newest toast at the top of the stack

Resolves #172

### CODE REVIEW

File: src/ToastProvider.js [hunk 1]
```diff
@@ -62,6 +62,8 @@ type Props = {
   // A convenience prop; the duration of the toast transition, in milliseconds.
   // Note that specifying this will override any defaults set on individual children Toasts.
   transitionDuration: number,
+  // have the newest toast at the top of the stack 
+  newestOnTop: boolean
```
Comment (jossmac) [inline src/ToastProvider.js:66]: Please add docs to the [README](https://github.com/jossmac/react-toast-notifications#toastprovider-props).

**Review by jossmac — CHANGES_REQUESTED** [review]

File: src/ToastProvider.js (hunk 1 above)
Comment (jossmac) [reply src/ToastProvider.js:66]: Please update [types](https://github.com/jossmac/react-toast-notifications/blob/master/index.d.ts#L44).

Comment (jossmac) [reply src/ToastProvider.js:66]: Please sort alphabetically, like the other props.

**Review by jossmac — CHANGES_REQUESTED** [review]
Looks good. A few things:
- Add docs to the [README](https://github.com/jossmac/react-toast-notifications#toastprovider-props).
- Update [types](https://github.com/jossmac/react-toast-notifications/blob/master/index.d.ts#L44).
- Sort alphabetically, like other props

File: src/ToastProvider.js [hunk 2]
```diff
@@ -55,13 +55,15 @@ type Props = {
   children: Node,
   // Component replacement object
   components: Components,
+  // have the newest toast at the top of the stack
```
Comment (jossmac) [inline src/ToastProvider.js:58]:
```suggestion
  // When true, insert new toasts at the top of the stack
```

File: README.md [hunk 3]
```diff
@@ -55,6 +55,7 @@ For brevity:
 | autoDismiss `boolean`                  | Default: `false`. Whether or not to dismiss the toast automatically after a timeout.       |
 | children `Node`                        | Required. Your app content.                                                                |
 | components `{ ToastContainer, Toast }` | Replace the underlying components.                                                         |
+| newestOnTop `boolean`                  | Default `false`. Whether to have the newest toast at the top or bottom of the toastStack.  |
```
Comment (jossmac) [inline README.md:58]:
```suggestion
| newestOnTop `boolean`                  | Default `false`. When true, insert new toasts at the top of the stack.  |
```

File: src/ToastProvider.js [hunk 4]
```diff
@@ -55,13 +55,15 @@ type Props = {
...
+  // have the newest toast at the top of the stack 
+  newestOnTop: boolean,
   // Where, in relation to the viewport, to place the toasts
   placement: Placement,
   // Which element to attach the container's portal to, defaults to the `body`.
   portalTargetSelector?: string,
   // A convenience prop; the duration of the toast transition, in milliseconds.
   // Note that specifying this will override any defaults set on individual children Toasts.
-  transitionDuration: number,
+  transitionDuration: number
```
Comment (jossmac) [inline src/ToastProvider.js:66]:
```suggestion
  transitionDuration: number,
```

File: src/ToastProvider.js [hunk 5]
```diff
@@ -77,8 +79,9 @@ export class ToastProvider extends Component<Props, State> {
     autoDismiss: false,
     autoDismissTimeout: 5000,
     components: defaultComponents,
+    newestOnTop: false,
     placement: 'top-right',
-    transitionDuration: 220,
+    transitionDuration: 220
```
Comment (jossmac) [inline src/ToastProvider.js:84]:
```suggestion
    transitionDuration: 220,
```

File: src/ToastProvider.js (hunk 2 above)
Comment (ochomoswill) [reply src/ToastProvider.js:58]: Cool. That sounds good.

**Review by jossmac — APPROVED** [review]

**Comment by jossmac** [comment]
Available from `2.5.0`

**Pull request merged**